File: packages_py/runtime_template_resolver/examples/basic_usage.py
"""
Basic usage examples for runtime_template_resolver.
"""
from runtime_template_resolver import resolve

def example_basic_usage():
    print("--- Basic Usage ---")
    context = {
        "user": {
            "name": "Alice",
            "role": "admin"
        },
        "env": "production"
    }
    
    # Mustache style
    template1 = "Hello {{user.name}}, welcome to {{env}}."
    print(f"Template: {template1}")
    print(f"Result:   {resolve(template1, context)}")
    
    # Dot path style
    template2 = "Role: $.user.role"
    print(f"Template: {template2}")
    print(f"Result:   {resolve(template2, context)}")
    
def example_defaults():
    print("\n--- Defaults ---")
    context = {}
    
    # Default value
    template = "User: {{user.name|\"Guest\"}}"
    print(f"Template: {template}")
    print(f"Result:   {resolve(template, context)}")
    
def example_arrays():
    print("\n--- Arrays ---")
    context = {"items": ["apple", "banana"]}
    
    template = "First item: {{items[0]}}"
    print(f"Template: {template}")
    print(f"Result:   {resolve(template, context)}")

if __name__ == "__main__":
    example_basic_usage()
    example_defaults()
    example_arrays()

File: packages_mjs/runtime-template-resolver/examples/basic-usage.mjs
/**
 * Basic usage examples for runtime-template-resolver.
 * Run with: node examples/basic-usage.mjs
 */
import { resolve } from '../dist/index.js'; // Assuming build output

// Mock implementation if dist unavailable (for development without build)
// import { resolve } from '../src/index.js'; 
// Note: importing ts directly requires loader, relying on dist for standard example

function exampleBasicUsage() {
    console.log("--- Basic Usage ---");
    const context = {
        user: {
            name: "Alice",
            role: "admin"
        },
        env: "production"
    };

    // Mustache style
    const template1 = "Hello {{user.name}}, welcome to {{env}}.";
    console.log(`Template: ${template1}`);
    console.log(`Result:   ${resolve(template1, context)}`);

    // Dot path style
    const template2 = "Role: $.user.role";
    console.log(`Template: ${template2}`);
    console.log(`Result:   ${resolve(template2, context)}`);
}

function exampleDefaults() {
    console.log("\n--- Defaults ---");
    const context = {};

    // Default value
    const template = 'User: {{user.name|"Guest"}}';
    console.log(`Template: ${template}`);
    console.log(`Result:   ${resolve(template, context)}`);
}

function exampleArrays() {
    console.log("\n--- Arrays ---");
    const context = { items: ["apple", "banana"] };

    const template = "First item: {{items[0]}}";
    console.log(`Template: ${template}`);
    console.log(`Result:   ${resolve(template, context)}`);
}

// Check if resolve is imported correctly (dist might be missing if not built)
try {
    exampleBasicUsage();
    exampleDefaults();
    exampleArrays();
} catch (e) {
    console.error("Error running examples. Make sure package is built (npm run build).");
    console.error(e);
}
